upcloud_api/cloud_manager/storage_mixin.py

```diff
@@ -184,33 +184,37 @@ def create_storage_import(
         Creates an import task to import data into an existing storage.
         Source types: http_import or direct_upload.
         """
+        if source not in ["http_import", "direct_upload"]:
+            raise Exception("invalid storage import source: %s", source)
+
         url = f'/storage/{storage}/import'
         body = {'storage_import': {'source': source}}
         if source_location:
             body['storage_import']['source_location'] = source_location
         res = self.api.post_request(url, body)
         return StorageImport(**res['storage_import'])
 
-    def upload_file_for_storage_import(self, storage_import, file):
+    def upload_file_for_storage_import(self, storage_import: StorageImport, file: Path, timeout: int = 600):
```

Comment:
Allowing only a Path is not probably what you want. See https://docs.python.org/3/library/os.html#os.PathLike - that would allow a string too.